src/woo_publications/publications/admin.py

```diff
@@ -223,6 +225,20 @@ def save_model(
         is_published = new_status == PublicationStatusOptions.published
         is_revoked = new_status == PublicationStatusOptions.revoked
 
+        if not change:
+            retention_ic = get_retention_informatie_category(
+                form.cleaned_data["informatie_categorieen"]
+            )
+
+            if retention_ic:
```

Comment:
I think it's better to tap into the `save_related` method - you still have the `change` argument to see if it was a create or change, but you have the benefit that you can query `publication.informatie_categorieen.all()` and that will have been saved in the database already, so you don't need to mess around with `form.cleaned_data`

I think I'd then indeed opt for an `assert ic is not None`, with a comment saying that the information categories for a publication require at least one entry according to the model definition so we know where the assumption is coming from.